src/wasm/literal.cpp

```diff
@@ -33,35 +33,60 @@ Literal::Literal(const uint8_t init[16]) : type(Type::v128) {
   memcpy(&v128, init, 16);
 }
 
-Literal::Literal(const Literal& other) { *this = other; }
+Literal::Literal(const Literal& other) : type(other.type), v128() {
+  if (type.isException()) {
+    auto* otherExn = other.exn;
+    exn = otherExn != nullptr
+            ? new ExceptionPackage(otherExn->event, otherExn->values)
+            : nullptr;
+  } else if (type.isFunction()) {
+    func = other.func;
+  } else {
+    TODO_SINGLE_COMPOUND(type);
+    switch (type.getBasic()) {
+      case Type::i32:
+      case Type::f32:
+        i32 = other.i32;
+        break;
+      case Type::i64:
+      case Type::f64:
+        i64 = other.i64;
+        break;
+      case Type::v128:
+        memcpy(&v128, other.v128, 16);
+        break;
+      case Type::none:
+        break;
+      case Type::externref:
+      case Type::anyref:
+      case Type::eqref:
+        break; // null
+      case Type::funcref:
+      case Type::i31ref:
+      case Type::exnref:
+      case Type::unreachable:
+        WASM_UNREACHABLE("unexpected type");
+    }
+  }
+}
+
+Literal::Literal(ExceptionPackage& exn_) : type(Type::exnref) {
+  exn = new ExceptionPackage(exn_);
+}
+
+Literal::~Literal() {
+  if (type.isException() && exn != nullptr) {
+    delete exn;
+    exn = nullptr;
+  }
```

Comment:
...and the `delete` here seemed like it'd be safer than a smart pointer in a dumb struct, but didn't help.